### ISSUE TITLE
plugin does not work correctly with oxlint

### ISSUE THREAD

**Issue by camc314** [schoero/eslint-plugin-better-tailwindcss#321]
### Description

Oxlint maintainer here!

I originally thought this was a bug in oxlint, but now I'm not so sure.

Oxlint makes use of multiple threads, this means that the order in which files are linted is not consistent, this means that rules will run on different files in different orders (basically) every time.

If we have code such as:
```
const ctx = createTestContext();
    ctx.options = {
      ...ctx.options,
      entryPoint: "styles.css"
    };

    const tailwindContext = await createTailwindContext(async(ctx));
    const options = {
      collapse: true,
      logicalToPhysical: true,
      rem: 16
    };

    const callA = ["mb-4", "text-4xl"];
    const callB = ["text-red-500", "mb-4"];
    const target = ["text-red-500", "h-4", "w-4"];

    expect(tailwindContext.canonicalizeCandidates(target, options)).toEqual(["text-red-500", "size-4"]);

    tailwindContext.canonicalizeCandidates(callA, options);
    tailwindContext.canonicalizeCandidates(callB, options);

    expect(tailwindContext.canonicalizeCandidates(target, options)).toEqual(["text-red-500", "size-4"]);
```

I would expect both expect calls to pass, but the second fails:

```
 FAIL  src/tailwindcss/canonical-classes.order-sensitivity.test.ts > canonicalizeCandidates order sensitivity > should not change canonicalization based on prior calls
AssertionError: expected [ 'text-red-500', 'h-4', 'w-4' ] to deeply equal [ 'text-red-500', 'size-4' ]

- Expected
+ Received

  [
    "text-red-500",
-   "size-4",
+   "h-4",
+   "w-4",
  ]

 ❯ src/tailwindcss/canonical-classes.order-sensitivity.test.ts:42:69
     40|     tailwindContext.canonicalizeCandidates(callB, options);
     41| 
     42|     expect(tailwindContext.canonicalizeCandidates(target, options)).toEqual(["text-red-500", "size-4"]);
       |                                                                     ^
     43|   });
     44| });
```

Please take a look at this branch with the failing test here:

https://github.com/camc314/eslint-plugin-better-tailwindcss/commit/b22160e835b46a34826cbad4d0812d6f4c3246d6

Refs: https://github.com/oxc-project/oxc/issues/19034

### Flavor

JSX

### Code Input

```tsx
n/a
```

### Expected Behavior

```tsx
n/a
```

### Actual Behavior

```tsx
n/a
```

### ESLint Log

```shell
n/a
```

### ESLint Config

```typescript
n/a
```

### Versions

n/a

**Comment by deadcoder0904** [comment]
@camc314 In the meantime, is there a way to use tailwind canonical classes plugin with oxlint?
